aggregation/simple/src/main/java/com/spotify/heroic/aggregation/simple/RatePerSecondInstance.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2019 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.heroic.aggregation.simple
+
+import com.google.common.collect.ImmutableSet
+import com.google.common.util.concurrent.AtomicDouble
+import com.spotify.heroic.aggregation.AbstractBucket
+import com.spotify.heroic.aggregation.BucketAggregationInstance
+import com.spotify.heroic.aggregation.DoubleBucket
+import com.spotify.heroic.metric.Metric
+import com.spotify.heroic.metric.MetricType
+import com.spotify.heroic.metric.Point
+
+
+
+/*
+    Calculate the rate per second of cumulative counters. When the counter is reset
+    because of an instance restart the interval will be skipped.
+ */
+data class RatePerSecondInstance(
+    override val size: Long,
+    override val extent: Long
+) : BucketAggregationInstance<RatePerSecondInstance.RateBucket>(size,
```

Comment:
@hexedpackets this is the part I read about the buckets needing to be thread safe https://github.com/spotify/heroic/blob/092c340c30460129840fb8511c825fd773822ca4/heroic-component/src/main/java/com/spotify/heroic/aggregation/BucketAggregationInstance.kt#L43